pandas/tests/frame/methods/test_between_time.py

```diff
@@ -207,3 +208,106 @@ def test_between_time_datetimeindex(self):
         tm.assert_frame_equal(result, expected)
         tm.assert_frame_equal(result, expected2)
         assert len(result) == 12
+
+    # GH40245
+    @pytest.mark.parametrize("inc_start", (True, False))
+    @pytest.mark.parametrize("inc_end", (True, False))
+    def test_between_time_warn(self, inc_start, inc_end, frame_or_series):
+        rng = date_range("1/1/2000", "1/5/2000", freq="5min")
+        ts = DataFrame(np.random.randn(len(rng), 2), index=rng)
+        if frame_or_series is not DataFrame:
+            ts = ts[0]
+
+        stime = time(0, 0)
+        etime = time(1, 0)
+
+        match = (
+            "`include_start` and `include_end` "
+            "are deprecated infavour of `inclusive`."
+        )
+        with tm.assert_produces_warning(FutureWarning, match=match):
+            filtered = ts.between_time(stime, etime, inc_start, inc_end)
+
+        # new new new
+        exp_len = 13 * 4 + 1
+        if not inc_start:
+            exp_len -= 5
+        if not inc_end:
+            exp_len -= 4
+
+        assert len(filtered) == exp_len
```

Comment:
This test is testing the deprecated args. The previous user function test has been amended to test the new `inclusive` arg. I would be inclined to test here that the methods are equivalent:

1) meth() == meth(inclusive="both")
2) meth(inc_start=False) == meth(inclusive="right")
3) meth(inc_end=False) == meth(inclusive="left")
4) meth(inc_start=False, inc_end=False) == meth(inclusive="neither")
5) meth(inc_start=True, inc_end=True) == meth(inclusibe="both")

can be parametrized and should raise a warning but return equivalent result.

Reply:
Implementing this.